src/features/Students/StudentsPage/index.jsx

```diff
@@ -17,16 +17,19 @@ const StudentsPage = () => {
 
   useEffect(() => {
     if (Object.keys(selectedInstitution).length > 0) {
-      dispatch(fetchStudentsData(selectedInstitution.id, initialPage));
+      dispatch(fetchStudentsData(selectedInstitution.id, currentPage, stateStudents.filters));
     }
 
     return () => {
       dispatch(resetStudentsTable());
+      dispatch(updateFilters({}));
     };
-  }, [selectedInstitution, dispatch]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [selectedInstitution, dispatch, currentPage]); // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
-    dispatch(fetchStudentsData(selectedInstitution.id, currentPage, stateStudents.filters));
+    if (Object.keys(selectedInstitution).length > 0) {
+      dispatch(fetchStudentsData(selectedInstitution.id, currentPage, stateStudents.filters));
+    }
```

Comment:
Please correct me if I'm wrong but this state can be removed completed,  the previous state do this same thing, also this could drive into unexpected behaviors like left stuff in the state for filters

Reply:
@01001110J thanks!